auth_proxy/proxy/flask.py

```diff
@@ -36,30 +36,32 @@ class FlaskClient(Client):
         'Procedure',
     ]
 
+    SECURITY_ARG_NAME = '_security'
+
     def __init__(self, url, orig):
         self.url = url
         self.orig = orig
 
     def request(self):
         """ @inherit
         """
-        args = list(self.orig.args.items())
+        original_args = list(self.orig.args.items())
 
         # We don't care what the client thinks it should be able to see
-        args = [arg for arg in args if arg[0] != '_security']
+        stripped_args = self._strip_security_args(original_args)
 
         # Update the query params
         path = self.orig.view_args.get('path').split('/')
+
         if len(path) == 1:
-            args.append(('_security', self._scope_security()))
-            args.append(('_security', self._patient_security()))
+            self._scope_security(stripped_args)
```

Comment:
In general, I'd lean toward functions that return updated values, rather than mutating their arguments. This functional approach makes things easier to test (not that we have a whole lot of tests). 

(Alternatively, if you want to make functions mutate their args, best to name them in some way that makes this clear.

Reply:
Noted!